lib/std/process.zig

```diff
@@ -1162,10 +1162,16 @@ pub fn totalSystemMemory() TotalSystemMemoryError!usize {
         .linux => {
             return totalSystemMemoryLinux() catch return error.UnknownTotalSystemMemory;
         },
-        .freebsd => {
+        .freebsd, .netbsd, .openbsd, .dragonfly, .macos => {
             var physmem: c_ulong = undefined;
             var len: usize = @sizeOf(c_ulong);
-            os.sysctlbynameZ("hw.physmem", &physmem, &len, null, 0) catch |err| switch (err) {
+            var name = if (.builtin.os.tag == .macos)
+                "hw.memsize"
+            else if (.builtin.os.tag == .netbsd)
+                "hw.physmem64"
+            else
+                "hw.physmem";
+            os.sysctlbynameZ(name, &physmem, &len, null, 0) catch |err| switch (err) {
```

Comment:
Please note, there is no leading `.` in `builtin`. `builtin` is a standard import and not an enum variant. `name` doesn't need to be mutable. Also, a `switch` would be better here in my opinion like so:

```suggestion
            const name = switch (builtin.os.tag) {
                .macos => "hw.memsize"
                .netbsd => "hw.physmem64",
                else => "hw.physmem",
            };
            os.sysctlbynameZ(name, &physmem, &len, null, 0) catch |err| switch (err) {
```

Reply:
fair